storm-core/src/jvm/backtype/storm/scheduler/resource/ResourceAwareScheduler.java

```diff
@@ -28,94 +32,270 @@
+        this.initialize(topologies, cluster);
+        //logs everything that is currently scheduled and the location at which they are scheduled
+        LOG.info("Cluster scheduling:\n{}", ResourceUtils.printScheduling(cluster, topologies));
+        //logs the resources available/used for every node
+        LOG.info("Nodes:\n{}", this.nodes);
+        //logs the detailed info about each user
+        for (User user : this.getUserMap().values()) {
+            LOG.info(user.getDetailedInfo());
+        }
 
-        ResourceAwareStrategy RAStrategy = new ResourceAwareStrategy(cluster, topologies);
-        LOG.debug(printScheduling(cluster, topologies));
+        ISchedulingPriorityStrategy schedulingPrioritystrategy = null;
+        while (true) {
 
-        for (TopologyDetails td : topologies.getTopologies()) {
-            String topId = td.getId();
-            Map<WorkerSlot, Collection<ExecutorDetails>> schedulerAssignmentMap;
-            if (cluster.getUnassignedExecutors(td).size() > 0) {
-                LOG.debug("/********Scheduling topology {} ************/", topId);
-
-                schedulerAssignmentMap = RAStrategy.schedule(td);
-
-                double requestedMemOnHeap = td.getTotalRequestedMemOnHeap();
-                double requestedMemOffHeap = td.getTotalRequestedMemOffHeap();
-                double requestedCpu = td.getTotalRequestedCpu();
-                double assignedMemOnHeap = 0.0;
-                double assignedMemOffHeap = 0.0;
-                double assignedCpu = 0.0;
-
-                if (schedulerAssignmentMap != null) {
-                    try {
-                        Set<String> nodesUsed = new HashSet<String>();
-                        int assignedWorkers = schedulerAssignmentMap.keySet().size();
-                        for (Map.Entry<WorkerSlot, Collection<ExecutorDetails>> workerToTasksEntry : schedulerAssignmentMap.entrySet()) {
-                            WorkerSlot targetSlot = workerToTasksEntry.getKey();
-                            Collection<ExecutorDetails> execsNeedScheduling = workerToTasksEntry.getValue();
-                            RAS_Node targetNode = RAStrategy.idToNode(targetSlot.getNodeId());
-                            targetNode.assign(targetSlot, td, execsNeedScheduling, cluster);
-                            LOG.debug("ASSIGNMENT    TOPOLOGY: {}  TASKS: {} To Node: {} on Slot: {}",
-                                    td.getName(), execsNeedScheduling, targetNode.getHostname(), targetSlot.getPort());
-                            if (!nodesUsed.contains(targetNode.getId())) {
-                                nodesUsed.add(targetNode.getId());
+            if (schedulingPrioritystrategy == null) {
+                try {
+                    schedulingPrioritystrategy = (ISchedulingPriorityStrategy) Utils.newInstance((String) this.conf.get(Config.RESOURCE_AWARE_SCHEDULER_PRIORITY_STRATEGY));
+                } catch (RuntimeException e) {
```

Comment:
well this is to prevent the strategy from crashing nimbus potentially frequently if there is a bug in the priority strategy.  This way nimbus will be still up and users can view their topologies but no scheduling will be done for new incoming topologies